### PR TITLE
[BUG] Carousel pagination breaks after change in slides-per-page (Issue/3551)

### LINKED ISSUE #3551: [BUG] Carousel pagination breaks after change in slides-per-page
When the `slidesPerPage` prop changes dynamically (like its the case when using it as a breakpoint customizable value), the pagination will break after this change. Only happens when `focusOnCenterSlide` prop is used. Likely because newly introduced `getPageCount` function isn't used in the `updateAmountOfPages` function.

### JS Framework

- All

### Porsche Design System version

- 3.20.0

### On which device(s) you can see the bug?

All

### In which browsers you can see the bug?

All

### Reproduction Example

- https://stackblitz.com/edit/nxbess?file=index.html

### Steps to reproduce the behavior

1. Go to https://stackblitz.com/edit/nxbess?file=index.html
2. Resize viewport of example to show 2 slides per page
3. Navigate through slides
4. See broken pagination

### Expected behavior

- Pagination works after window resize

### Notes

Carousel has to be used with `focusOnCenterSlide` for the problem to occur. Introduced with https://github.com/porsche-design-system/porsche-design-system/issues/3438

Probably because `renderPagination` doesn't use `this.getPageCount()` here:

```
  private updateAmountOfPages = (): void => {
    this.amountOfPages = getAmountOfPages(
      this.slides.length,
      // round to sanitize floating numbers
      this.slidesPerPage === 'auto' ? 1 : Math.round(getCurrentMatchingBreakpointValue(this.slidesPerPage))
    );
    renderPagination(this.paginationEl, this.amountOfPages, this.splide?.index || 0, this.splide);
  };
```

### PR DESCRIPTION
**Pull request by o-hook**
### Pull Request Checklist

- [x] I have read and accept the [Contributor License Agreement](https://opensource.porsche.com/docs/cla)

#### References

- Preview: https://designsystem.porsche.com/issue/3551

#### Scope

Short description of the pull request scope

#### Resolved Issue

Resolves #3551

### CODE REVIEW

File: packages/components-js/tests/e2e/specs/carousel.e2e.ts [hunk 1]
```diff
@@ -610,6 +634,29 @@ test.describe('viewport change', () => {
     await waitForStencilLifecycle(page);
     expect(await pagination.evaluate((el) => el.children.length)).toBe(4);
   });
+
+  test('should update pagination for BreakpointCustomizable in focusOnCenterSlide mode', async ({ page }) => {
```
Comment (henrijoss) [inline packages/components-js/tests/e2e/specs/carousel.e2e.ts:638]: The problem here wasn't the breakpoint customisable but rather the dynamic change to the `slidesPerPage`. Do would be enough to just change this value dynamically.

File: packages/components/CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +21,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0),
 
 #### Fixed
 
+- `Carousel`: rendering of pagination in case of `focusOnCenterSlide` prop is set
```
Comment (henrijoss) [inline packages/components/CHANGELOG.md:24]:
```suggestion
- `Carousel`: dynamic change in `slidesPerPages` when using `focusOnCenterSlide` prop breaks pagination
```

File: packages/components-js/tests/e2e/specs/carousel.e2e.ts (hunk 1 above)
Comment (o-hook) [reply packages/components-js/tests/e2e/specs/carousel.e2e.ts]: Plz re-check :-)

File: packages/components-js/tests/e2e/specs/carousel.e2e.ts [hunk 3]
```diff
@@ -229,6 +229,24 @@ test('should update pagination on next button clicks', async ({ page }) => {
   expect(await getCssClasses(bullet3)).toBe('bullet');
 });
 
+test('should update pagination in focusOnCenterSlide mode if slidesPerPage changed dynamically', async ({ page }) => {
```
Comment (henrijoss) [inline packages/components-js/tests/e2e/specs/carousel.e2e.ts:232]: Maybe would be good to check if the pagination also still works and not only if the amount if correct?

**Review by henrijoss — APPROVED** [review]

**Pull request merged**